draft-ietf-tls-tls13.md

```diff
@@ -2312,7 +2320,47 @@ the existing behavior for SignedCertificateTimestamps {{?RFC6962}},
 and is more easily extensible in the handshake state machine.
 
 
-## Server Parameters Messages
+### Hello Finished
+
+The "hello_finished" extension is used to carry a Finished.verify_data
+({{finished}}) in the ClientHello. The reason for this structure rather
+than sending the Finished as a separate message is for backward
+compatibility: the client can send a "hello_finished" extension
+to servers which may not support TLS 1.3 without causing them to
+fail the handshake. This allows for mixed server configurations which
+concurrently support TLS 1.3 (including PSK resumption but not 0-RTT)
+and TLS 1.2 without creating interoperability problems.
+
+The "hello_finished" extension MUST be sent if the client offers a
+"pre_shared_key" extension and MUST NOT be sent otherwise. The Hash
+used in the computation is that associated with the indicated PSK.
+Servers MUST NOT send this extension. This extension MUST be the
```

Comment:
In general I'm not super-keen on ordering requirements within the extensions block, "but this one has a good reason so it's okay".